metakb/transform/pmkb.py

```diff
@@ -0,0 +1,454 @@
+        """
+        symbol = variant['gene']['name']
+        invalid_keys = self.invalid_keys['genes']
+        if symbol in invalid_keys:
+            return []
+
+        gene_descriptors = self.transformed['gene_descriptors']
+        vod = gene_descriptors.get(symbol)
+        if vod:
+            return [vod]
+
+        response = self.gene_normalizer.search_sources(symbol, incl='hgnc')
+        match = response['source_matches'][0]
+        if match['match_type'] == 0:
+            logger.warning(f"Gene normalization of {symbol} failed.")
+            invalid_keys.add(symbol)
+            return []
+
+        normalized_id = match['records'][0].concept_id
+
+        vod = schemas.ValueObjectDescriptor(
+            id=f"pmkb.normalize.gene:{symbol}",
+            type="GeneDescriptor",
+            label=symbol,
+            value=schemas.Gene(id=normalized_id)
+        ).dict(exclude_none=True)
+
+        gene_descriptors[symbol] = vod
+        return [vod]
+
+    def _get_variant_descriptors(self, statement, variant, gene_id):
+        """Fetch variant descriptors.
+        :param Dict statement: PMKB statement object
+        :param Dict variant: PMKB variant object
+        :param str gene_id: identifier for gene_context field
+        :return: List (len == 1) containing VOD of normalized match, or
+            empty list if normalization fails
+        """
+        label = variant['name']
+        invalid_keys = self.invalid_keys['variants']
+        if label in invalid_keys:
+            return []
+
+        vod = self.transformed['variation_descriptors'].get(label)
+        if vod:
+            return [vod]
+
+        validations = self.variant_to_vrs.get_validations(label)
+        response = self.variant_normalizer.normalize(label, validations,
+                                                     self.amino_acid_cache)
+        if not response:
+            logger.warning(f"Variant normalization of {label} failed.")
+            invalid_keys.add(label)
+            return []
+
+        vod = schemas.VariationDescriptor(
+            id=f"pmkb.variant:{variant['id']}",
+            type="VariationDescriptor",
+            label=label,
+            value_id=response.value_id,
+            value=response.value,
+            gene_context=f"pmkb.gene:{variant['gene']['name']}",
+            molecule_context=response.molecule_context,
+            structural_type=response.structural_type,
+            ref_allele_seq=response.ref_allele_seq,
+        ).dict(exclude_none=True)
+
+        # TODO extensions?
+        assoc_with = []
+        cosmic_id = variant.get('cosmic_id')
+        if cosmic_id:
+            assoc_with.append(f'{schemas.XrefSystem.COSMIC.value}:{cosmic_id}')
+        ens_id = variant.get('ensembl_id')
+        if ens_id:
+            assoc_with.append(f'{schemas.XrefSystem.ENSEMBL.value}:{ens_id}')
+        if assoc_with:
+            vod['extensions'] = schemas.Extension(
+                name="associated_with",
```

Comment:
I might be wrong on this
```suggestion
                name="xrefs",
```

Reply:
I'm gonna ask Alex about this to confirm but I think xrefs makes sense here